packages/@lwc/engine/src/framework/template.ts

```diff
@@ -72,24 +72,25 @@ export interface Template {
         shadowAttribute: string;
     };
 }
-const EmptySlots: SlotSet = create(null);
 
 function validateSlots(vm: VM, html: any) {
     if (process.env.NODE_ENV === 'production') {
         // this method should never leak to prod
         throw new ReferenceError();
     }
-    const { cmpSlots = EmptySlots } = vm;
+    const { children } = vm;
     const { slots = EmptyArray } = html;
-    for (const slotName in cmpSlots) {
-        // eslint-disable-next-line no-production-assert
-        assert.isTrue(
-            isArray(cmpSlots[slotName]),
-            `Slots can only be set to an array, instead received ${toString(
-                cmpSlots[slotName]
-            )} for slot "${slotName}" in ${vm}.`
-        );
-
+    const slotNameRecord: Record<string, 1> = create(null);
```

Comment:
a little bit more work to validate that the provided slot name is a valid one because we don't keep the mapping anymore in memory.